src/Redux/mission/Mission.js

```diff
@@ -0,0 +1,65 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+
+const SHOW_MISSION = "missions/missions/SOW_MISSION";
+const SHOW_STATUS = "missions/missions/SHOW_STATUS";
+const LEAVE_MISSIONS = "missions/missions/leaveMissions";
+
+const url = "https://api.spacexdata.com/v3/missions";
+const initialState = { missionArray: [] };
+
+export const showMission = createAsyncThunk(
+  SHOW_MISSION,
+  async (args, { dispatch }) => {
+    const response = await fetch(url);
+    const data = await response.json();
+    const missions = Object.keys(data).map((key) => {
+      const mission = data[key];
+      return {
+        id: key,
+        ...mission,
+      };
+    });
+    dispatch({
+      type: SHOW_MISSION,
+      payload: missions,
+    });
+    return missions;
+  },
+);
+
+export const showStatus = (id) => (dispatch) => {
+  dispatch({
+    type: SHOW_STATUS,
+    payload: id,
+  });
+};
+
+export const LeaveMission = (id) => (dispatch) => {
+  dispatch({ type: LEAVE_MISSIONS, payload: id });
+};
+
+const missionReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case SHOW_MISSION:
+      return {
+        ...state,
+        missionArray: action.payload,
+      };
+    case SHOW_STATUS:
+      return {
+        missionArray: state.missionArray.map((mission) => (mission.mission_id === action.payload
+          ? { ...mission, reserved: true }
+          : mission)),
+      };
```

Comment:
- [ ] You seem to have implemented your reservation status wrongly. As can see a reserved mission should have the `reserved` attribute set to `true`, but in your case, it is set to `false`, and vice versa. Kindly rectify that.

  ![ddp](https://user-images.githubusercontent.com/54848363/193020059-5820e1b3-b8e3-449e-98cf-b33a78215019.jpg)

Reply:
hey @DammyShittu ,
thanks for this, but I think you console log to the wrong destination see the snaps below.. but yeah I made some changes.
## 1 see the mission below ⏬ 
![image](https://user-images.githubusercontent.com/104095543/193206559-c9a3567d-1331-419e-b2b4-77e199dfd8ba.png)

## 2 see the profile below.  ⏬ 

![new1](https://user-images.githubusercontent.com/104095543/193206788-0871803f-88da-4fb2-9ec8-fb9ca58251bb.png)